### PR TITLE
fix(vcs): recursively handle submodules when scanning for files

### LINKED ISSUE #1097: Garden won't find modules that are git submodules
Hey there.

I have a simple meta-repository: The garden project level configuration resides in the root directory, modules are stored in git submodules that are mounted in this root. 

_meta # meta git repository
├── submodule
│   ├── Dockerfile
│   └── garden.yml
└── not-a-submodule
│   └── garden.yml
└── garden.yml

When adding a garden module configuration in one of the submodules,  garden won't find them with `garden scan` from the root directory. Simply `cp submodule/garden.yml not-a-submodule/` will make garden pick up the configuration from the folder that's not a submodule. Adding the not-a-submodule directory to .gitignore doesn't change anything.

Any thoughts? :confused:

### LINKED ISSUE #1190: Unable to overlap .gitignore setting
## Bug

I can not manually include garden.yaml, when the project has own` .git`.

simplified folders structure

```
├── core
│  ├── .git
│  ├── garden.yml
└── garden.yml
```

where `core` is excluded from git.

```
# .gitignore

/core
```

```yaml
# core/garden.yaml
---
# some config here
```

```yaml
# ./garden.yaml
kind: Project
name: rp
modules:
  include:
    - core/garden.yaml
...
```

If try the same config with out `.git` folder in the projects- `garden scan` founds the configuration file.

### Current Behavior

```bash
garden scan               
✔ providers                 → Getting status... → Done
modules: []
```

### Additional context

I tried to put it in another folder, which is not in `.gitignore` and with out own git repo, but it produced:

```bash
$ garden scan
Error validating module poker-core (support/core/garden.yml): key .dockerfile must be a relative sub-path (may not contain '..' segments or be an absolute path)
```

```bash
# support/core/garden.yaml

---
kind: Module
description: Image for poker-core
type: container
name: poker-core
dockerfile: ../../core/docker/development/Dockerfile
```

For me it is a blocker now.

### Your environment

`garden version`
0.10.9
`kubectl version`
1.15.2
`docker version`
2.1.0.2

### PR DESCRIPTION
**Pull request by edvald**
**What this PR does / why we need it**:

This is implemented by explicitly checking for configured submodules,
and recursively scanning each submodule that passes the given ignores
and include/exclude filters.

See the added section in the Configuration Files guide for some notes
on the intended behavior.

**Which issue(s) this PR fixes**:

Fixes #1097
Closes #1190

**Special notes for your reviewer**:

Make sure the semantics described feel understandable and reasonable. Look at the added tests to see how it's intended to work. Please try to think of edge-cases I hadn't considered.

### CODE REVIEW

File: docs/using-garden/configuration-files.md [hunk 1]
```diff
@@ -399,6 +399,14 @@ name: my-project
 dotIgnoreFiles: [.gardenignore]
 ```
 
+#### Submodules
```
Comment (eysi09) [inline docs/using-garden/configuration-files.md]: Should this be `Git Submodules` so there's no confusion with e.g. Garden remote modules.

File: docs/using-garden/configuration-files.md [hunk 2]
```diff
@@ -399,6 +399,14 @@ name: my-project
...
+
+If you're using submodules in your project, please note the following:
+
+1. You may ignore submodules using .ignore files and include/exclude filters. If a submodule path _itself_ matches one that is ignored by your .ignore files or exclude filters, or if you specify include filters and the submodule path does not match one of them, the module will _not_ be scanned.
```
Comment (eysi09) [inline docs/using-garden/configuration-files.md]: Isn't this the same behaviour as for other directories?

Comment (edvald) [reply docs/using-garden/configuration-files.md]: It is in most ways, except the "_itself_" part (matching the path of the submodule, and not its contents). I can try to clarify.

**Comment by edvald** [comment]
Updated the docs.

**Comment by eysi09** [comment]
Noticed there's a test that's failing.

**Comment by Co0ker** [comment]
Hi, @edvald, @eysi09  
Its not working :(

image

garden.yml:
```
kind: Project
name: test-submodule
dotIgnoreFiles: [.gardenignore]
```
dir/garden.yml:
```
kind: Module
description: Test GIT submodule
type: container
name: dir
```

submodule/garden.yml:
```
kind: Module
description: Test GIT submodule
type: container
name: submodule
```

1st run `garden scan` - found 2 modules

run `git init` in dir `submodule` - found 2 modules

`commit` garden.ym in dir `submodule` - found 1 module

Garden: `0.11.5`
Git: `2.17.2 (Apple Git-113)`
